src/pubsub/mod_pubsub_db_rdbms.erl

```diff
@@ -228,49 +228,172 @@ del_node(Nidx) ->
+    DecodedOwners = [jid:to_lower(jid:from_binary(Owner)) ||
+                     Owner <- jsx:decode(Owners)],
+    #pubsub_node{nodeid = {Key, Name},
+                 id = mongoose_rdbms:result_to_integer(Nidx),
+                 type = Type,
+                 owners = DecodedOwners,
+                 options = DecodedOptions}.
+
+maybe_option_value_to_atom({access_model, Value}) ->
+    {access_model, binary_to_existing_atom(Value, utf8)};
+maybe_option_value_to_atom({publish_model, Value}) ->
+    {publish_model, binary_to_existing_atom(Value, utf8)};
+maybe_option_value_to_atom({notification_type, Value}) ->
+    {notification_type, binary_to_existing_atom(Value, utf8)};
+maybe_option_value_to_atom({node_type, Value}) ->
+    {node_type, binary_to_existing_atom(Value, utf8)};
+maybe_option_value_to_atom({send_last_published_item, Value}) ->
+    {send_last_published_item, binary_to_existing_atom(Value, utf8)};
+maybe_option_value_to_atom(Other) ->
+    Other.
 
 -spec find_nodes_by_key(Key :: mod_pubsub:hostPubsub() | jid:ljid()) ->
     [mod_pubsub:pubsubNode()].
 find_nodes_by_key(Key) ->
-    mod_pubsub_db_mnesia:find_nodes_by_key(Key).
+    SQL = mod_pubsub_db_rdbms_sql:select_nodes_by_key(encode_key(Key)),
+    {selected, Rows} = mongoose_rdbms:sql_query(global, SQL),
+    [decode_pubsub_node_row(Row) || Row <- Rows].
 
 
 -spec delete_node(Key :: mod_pubsub:hostPubsub() | jid:ljid(), Node :: mod_pubsub:nodeId()) -> ok.
 delete_node(Key, Node) ->
-    mod_pubsub_db_mnesia:delete_node(Key, Node).
+    SQL = mod_pubsub_db_rdbms_sql:delete_node(encode_key(Key), Node),
+    {updated, _} = mongoose_rdbms:sql_query(global, SQL),
+    ok.
 
 -spec get_subnodes(Key :: mod_pubsub:hostPubsub() | jid:ljid(), Node :: mod_pubsub:nodeId() | <<>>) ->
     [mod_pubsub:pubsubNode()].
 get_subnodes(Key, Node) ->
-    mod_pubsub_db_mnesia:get_subnodes(Key, Node).
-
--spec get_parentnodes(Key :: mod_pubsub:hostPubsub() | jid:ljid(), Node :: mod_pubsub:nodeId()) ->
-    [mod_pubsub:pubsubNode()].
-get_parentnodes(Key, Node) ->
-    mod_pubsub_db_mnesia:get_parentnodes(Key, Node).
+    SQL = mod_pubsub_db_rdbms_sql:select_subnodes(encode_key(Key), Node),
+    {selected, Rows} = mongoose_rdbms:sql_query(global, SQL),
+    [decode_pubsub_node_row(Row) || Row <- Rows].
 
 -spec get_parentnodes_tree(Key :: mod_pubsub:hostPubsub() | jid:ljid(), Node :: mod_pubsub:nodeId()) ->
     [{Depth::non_neg_integer(), Nodes::[mod_pubsub:pubsubNode(), ...]}].
 get_parentnodes_tree(Key, Node) ->
-    mod_pubsub_db_mnesia:get_parentnodes_tree(Key, Node).
+    find_nodes_with_parents(Key, [Node], 0, []).
+
+find_nodes_with_parents(_, [], _, Acc) ->
+    Acc;
+find_nodes_with_parents(_, _, 100, Acc) ->
+    ?WARNING_MSG("event=max_depth_reached, nodes=~p", [Acc]),
+    Acc;
+find_nodes_with_parents(Key, Nodes, Depth, Acc) ->
+    SQL = mod_pubsub_db_rdbms_sql:select_nodes_by_key_and_names_in_list_with_parents(
```

Comment:
What are we going to do one day when a proper function name + module name exceeds 100 characters alone? :D

Reply:
I don't know yet :)